crd/multitenancy/api/v1alpha1/podnetwork.go

```diff
@@ -40,8 +40,8 @@ type PodNetworkSpec struct {
 	// +kubebuilder:validation:Optional
 	// customer vnet guid
 	VnetGUID string `json:"vnetGUID,omitempty"`
-	// customer subnet id
-	SubnetResourceID string `json:"subnetResourceID,omitempty"`
+	// customer subnet guid
+	SubnetGUID string `json:"subnetGUID,omitempty"`
```

Comment:
I do not think we should remove the ARM resource id field. 
This is still the public identifier of the subnet, and how we'd find the scope subscription / resourcegroup / etc. 
the reason we do not include that for vnet is because the vnet arm path is already implicit on the subnet resource id.

Reply:
Restored ARM resource ID field for subnet, added subnetGUID as additional field.